service/history/historyEngine.go

```diff
@@ -2397,6 +2388,23 @@ func validateActivityScheduleAttributes(attributes *workflow.ScheduleActivityTas
 		}
 	}
 
+	validScheduleToClose := attributes.GetScheduleToCloseTimeoutSeconds() > 0
+	validScheduleToStart := attributes.GetScheduleToStartTimeoutSeconds() > 0
+	validStartToClose := attributes.GetStartToCloseTimeoutSeconds() > 0
+
+	if validScheduleToClose {
+		if !validScheduleToStart {
+			attributes.ScheduleToStartTimeoutSeconds = common.Int32Ptr(attributes.GetScheduleToCloseTimeoutSeconds())
+		}
+		if !validStartToClose {
+			attributes.StartToCloseTimeoutSeconds = common.Int32Ptr(attributes.GetScheduleToCloseTimeoutSeconds())
+		}
+	} else if validScheduleToStart && validStartToClose {
+		attributes.ScheduleToCloseTimeoutSeconds = common.Int32Ptr(attributes.GetScheduleToStartTimeoutSeconds() + attributes.GetStartToCloseTimeoutSeconds())
```

Comment:
and same here, only overwrite when ScheduleToCloseTimeoutSeconds is 0

Reply:
resolved